### DISCUSSION TITLE
Thread Dump when /cmi migratedatabase

### DISCUSSION THREAD

**Discussion by BiGUNMAN** [Zrips/CMI#2997] [Q&A]
**Question: **
When migrating database from sqlite to mysql, I get a thread dump on MC 1.12 with CMI version: 8.6.20.9

Dump: https://hastebin.com/etitupokaq.md

**Why are we migrating?**
Because CMI sqlite is randomly fails at saving data. Hastebin of an example: https://hastebin.com/ometivazar.css (players losing balances, etc).

_Playertime Reward Data Corruption (Playtime Data)_
We've also seen issues with data like `playtime rewards`. This is our playtime rewards file: https://hastebin.com/kemocesoxo.bash

Some users type /prewards, and it says they have to wait 25, or 40 minutes until their next reward. This reward is every 15 minutes!

**Comment by Zrips** [comment]
It could be nice to see full error log, as currently yours is stipped down by quite a bit and doesn't indicate exactly why its failed. I did managed to find some mentionings about issue with sqlite issues in general which could have resulted in something like this, but having actual full error log would help out.
For database migration, will make some improvements with next update, so it could handle as many users as we could throw at it without crashing, as this was a result of too many users being transfered to new database in one go.
